mosviz/cli.py

```diff
@@ -1,25 +1,97 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
 import sys
+import os
+import argparse
 
-from glue.main import main as glue_main
+from glue.app.qt import GlueApplication
+from glue.main import restore_session, get_splash, load_data_files, load_plugins
+from qtpy.QtCore import QTimer
+from qtpy import QtGui, QtWidgets
+
+from mosviz import __version__
 
 try:
     from glue.utils.qt.decorators import die_on_error
 except ImportError:
     from glue.utils.decorators import die_on_error
 
 
-@die_on_error("Error starting up MOSViz")
-def main(argv=sys.argv):
+MOSVIZ_ICON_PATH = os.path.abspath(
+    os.path.join(
+        os.path.abspath(__file__),
+        "..",
+        "data",
+        "resources",
+        "mosviz_icon.png"
+    )
+)
 
-    # Add startup item for MOSViz - this makes it so that the data gets
-    # auto-added to the canvas.
-    if '--startup=mosviz' not in argv:
-        argv.append('--startup=mosviz')
+MOSVIZ_SPLASH_PATH = os.path.abspath(
+    os.path.join(
+        os.path.abspath(__file__),
+        "..",
+        "data",
+        "resources",
+        "mosviz_splash.png"
+    )
+)
 
+
+@die_on_error("Error starting up MOSViz")
+def main(argv=sys.argv):
+    """
+    The majority of the code in this function was taken from start_glue() in main.py after a discussion with
+    Tom Robataille. We wanted the ability to get command line arguments and use them in here and this seemed
+    to be the cleanest way to do it.
+    """
     # Make sure the mosviz startup item is registered
     from .startup import mosviz_setup  # noqa
 
-    # Launch glue
-    glue_main(argv=argv)
+    parser = argparse.ArgumentParser()
+    parser.add_argument('data_files', nargs=argparse.REMAINDER)
+    args = parser.parse_known_args(argv[1:])
+
+    import glue
+    from glue.utils.qt import get_qapp
+    app = get_qapp()
+
+    # Splash screen
+    splash = get_splash()
+    splash.image = QtGui.QPixmap(MOSVIZ_SPLASH_PATH)
+    splash.show()
+
+    # Start off by loading plugins. We need to do this before restoring
+    # the session or loading the configuration since these may use existing
+    # plugins.
+    load_plugins(splash=splash)
+
+    datafiles = args[0].data_files
+
+    # # Show the splash screen for 2 seconds
+    timer = QTimer()
+    timer.setInterval(2000)
+    timer.setSingleShot(True)
+    timer.timeout.connect(splash.close)
+    timer.start()
+
+    data_collection = glue.core.DataCollection()
+    hub = data_collection.hub
+
+    splash.set_progress(100)
+
+    session = glue.core.Session(data_collection=data_collection, hub=hub)
+    ga = GlueApplication(session=session)
+    qapp = QtWidgets.QApplication.instance()
+    ga.setWindowTitle('MOSViz v{0}'.format(__version__))
+    qapp.setWindowIcon(QtGui.QIcon(MOSVIZ_ICON_PATH))
+    ga.setWindowIcon(QtGui.QIcon(MOSVIZ_ICON_PATH))
```

Comment:
I'm unfamiliar with this: are the `setWindowIcon` for the `GlueApplication` and the Qt Application instance really separate things?

Reply:
@nmearl : They are different in this way:
<img width="436" alt="screen shot 2018-09-12 at 10 05 41 am" src="https://user-images.githubusercontent.com/10345916/45430506-7679fb00-b673-11e8-8284-9ede38f38416.png">

Reply:
Is `isinstance(application, qapp.__class__)`? E.g. is the `GlueApplication` just a subclass of the `QApplication`?

Reply:
What I am getting at is, does the window icon need to be set in both places explicitly?

Reply:
It weird but `isinstance(application, qapp.__class__)` is `False`. `GlueApplication` is a `HubListener` subclass. `qapp.setWindowIcon` sets the icon on the MAC Doc and the `ga.setWindowIcon` sets the icon on the window title bar.

Reply:
That is super weird! Thanks for figuring that out.